File: reactivestreams/src/main/kotlin/reactivestreams/r2dbc/controller/UserController.java
package reactivestreams.r2dbc.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactivestreams.r2dbc.service.UserService;
import reactivestreams.r2dbc.service.response.UserResponse;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
@RestController
@RequestMapping("/user")
public class UserController {

    private final UserService userService;

    @GetMapping("/{userId}")
    public Mono<UserResponse> findUser(@PathVariable Long userId) {
        return userService.findById(userId);
    }

    @GetMapping("/name")
    public Mono<String> findNameWithContext() {
        String principal = (String) SecurityContextHolder
                .getContext()
                .getAuthentication()
                .getPrincipal();

        return Mono.just(principal);
    }

}

File: reactivestreams/src/main/kotlin/reactivestreams/coroutine/controller/UserController.kt
package reactivestreams.coroutine.controller

import lombok.RequiredArgsConstructor
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactivestreams.coroutine.service.UserService
import reactivestreams.coroutine.service.response.UserResponse
import reactor.core.publisher.Mono

@RequiredArgsConstructor
@RestController
@RequestMapping("/user")
class UserController(
    private val userService: UserService,
) {

    @GetMapping("/{userId}")
    suspend fun findUser(@PathVariable userId: Long): UserResponse {
        return userService.findById(userId)
    }

    @GetMapping("/name")
    fun findNameWithContext(): Mono<String> {
        val principal = SecurityContextHolder
            .getContext()
            .authentication
            .principal as String

        return Mono.just(principal)
    }

}